From 9efcc0f831be7b19fe5442517842730e869f262d Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Tue, 31 Mar 2020 10:06:25 +0300
Subject: [PATCH 2/5] Revert "IconPack: Fix setting summary after selecting
 icon pack"

This reverts commit 2ea7f5f4e444ba9834decd3efa03e3bde2ef5eaa.
---
 res/values/cr_strings.xml                     |  1 -
 .../settings/IconPackPreference.java          | 22 ++++++++-----------
 2 files changed, 9 insertions(+), 14 deletions(-)

diff --git a/res/values/cr_strings.xml b/res/values/cr_strings.xml
index 48aecba46..df09cfe25 100644
--- a/res/values/cr_strings.xml
+++ b/res/values/cr_strings.xml
@@ -68,5 +68,4 @@
     <string name="iconPackPackageDefault" translatable="false">""</string>
     <string name="icon_pack">Icon pack</string>
     <string name="icon_pack_summary">Select a custom icon pack</string>
-    <string name="icon_pack_default">Default</string>
 </resources>
diff --git a/src/com/android/launcher3/settings/IconPackPreference.java b/src/com/android/launcher3/settings/IconPackPreference.java
index 24affa1ac..46abd0b2a 100644
--- a/src/com/android/launcher3/settings/IconPackPreference.java
+++ b/src/com/android/launcher3/settings/IconPackPreference.java
@@ -33,7 +33,6 @@ import com.android.launcher3.R;
 public class IconPackPreference extends ListPreference {
 
     private final PackageManager pm;
-    private Context mContext;
 
     public IconPackPreference(Context context) {
         this(context, null);
@@ -45,44 +44,41 @@ public class IconPackPreference extends ListPreference {
 
     public IconPackPreference(Context context, AttributeSet attrs, int defStyleAttr) {
         super(context, attrs, defStyleAttr);
-        mContext = context;
         setLayoutResource(R.layout.preference_iconpack);
-        pm = mContext.getPackageManager();
+        pm = context.getPackageManager();
         init();
     }
 
     private void init() {
-        String currentPack = IconPackProvider.getCurrentIconPack(mContext);
+        String currentPack = IconPackProvider.getCurrentIconPack(getContext());
         if (currentPack.isEmpty()) {
-            setSummary(mContext.getString(R.string.icon_pack_default));
+            setNone();
         } else {
             try {
                 ApplicationInfo info = pm.getApplicationInfo(currentPack, 0);
                 setSummary(info.loadLabel(pm));
             } catch (PackageManager.NameNotFoundException e) {
-                setSummary(mContext.getString(R.string.icon_pack_default));
+                setNone();
                 persistString("");
             }
         }
     }
 
+    private void setNone() {
+        setSummary("None");
+    }
+
     @Override
     protected void onClick() {
         load();
         super.onClick();
     }
 
-    @Override
-    public void setValue(String value) {
-        super.setValue(value);
-        init();
-    }
-
     protected void load() {
         final Map<String, IconPackInfo> packages = loadAvailableIconPacks();
         String[] packageNames = new String[packages.size()+1];
         CharSequence[] labels = new CharSequence[packages.size()+1];
-        String defaultLabel = mContext.getString(R.string.icon_pack_default);
+        String defaultLabel = "None";
         packageNames[0] = "";
         labels[0] = defaultLabel;
         int i = 1;
-- 
2.17.1

